lib/default.nix

```diff
@@ -21,7 +21,7 @@ rec {
 
     lib.recursiveUpdate originalDrv {
       __nixpkgs-hammering-state = {
-        reports = originalDrv.__nixpkgs-hammering-state.reports or [] ++ reports;
+        reports = lib.unique originalDrv.__nixpkgs-hammering-state.reports or [] ++ reports;
```

Comment:
You will probably want to apply it to the rest of the line or duplicates might get introduced by the ultimate call of `addReports`.

Reply:
Right. I forgot my order of operations.